src/app/inicio/components.tsx

```diff
@@ -0,0 +1,160 @@
+import React, {useRef} from "react";
+import { userInputsArr, userInputStore, cicleShotsTypes, ShotsTypes } from '@/store/inputStore';
+
+interface ModelParameters{
+    value: string;
+    onChange: (event:any)=>void;
+}
+
+//model are the inputs from the user, these are part of the prompt
+export function Model({value, onChange}:ModelParameters){
+
+    return(
+        <>
+            <h1>Input something</h1>
+            <textarea name="" 
+                      id="input-model-alwaysActive" 
+                      onChange={onChange}
+                      className="text-black h-44 w-[40rem] rounded resize-none p-2" 
+            >
+            </textarea> <br />
+
+        </>
+    )
+}
+
+interface ResultTypes{
+    data: string | undefined | null;
+    value: string;
+    onChange: (event:any)=>void;
+}
+
+//results are GPT responses exemplified by the user
+export function Result({data, value, onChange}: ResultTypes){
+    const[userInputs, 
+        updateUserInput, 
+        updateUserResults, 
+        userResults] = userInputStore((state) => [state.userInputs, 
+                                                  state.updateUserInput, 
+                                                  state.updateUserResults, 
+                                                  state.userResults]);
+    const [updateUserInputArr] = userInputsArr((state)=> [state.updateUserInputArr]);
+
+    const updateShots = ()=>{    
+        let values:ShotsTypes = {input: userInputs, result:userResults};    
+        updateUserInputArr(values);
+        updateUserInput("");
+        updateUserResults("");
+    }
+    const inputRef:any = useRef(null);
+    const deleteText = ()=>{
+        if(inputRef.current != null){
+            inputRef.current.value = "";
+        }
+        if(document.getElementById('input-model-alwaysActive') != null){
+            (document.getElementById('input-model-alwaysActive') as HTMLInputElement).value = "";
+        }
```

Comment:
there is a better way to handle this without having to access the document or use ref by using controlled inputs.
Try reading about this, here is a page with general info as well.
https://react.dev/reference/react-dom/components/input
If you need some help or extra explanation, tell me and we can have call 🙂